math/mathcore/CMakeLists.txt

```diff
@@ -195,6 +195,21 @@ ROOT_STANDARD_LIBRARY_PACKAGE(MathCore
     ${MATHCORE_BUILTINS}
 )
 
+target_include_directories(MathCore PRIVATE ${Vc_INCLUDE_DIR})
+target_include_directories(MathCore PRIVATE ${VecCore_INCLUDE_DIRS})
+
+list(APPEND math_incl ${CMAKE_SOURCE_DIR}/math/mathcore/inc)
+if(mathmore)
+   target_include_directories(MathCore PRIVATE ${CMAKE_SOURCE_DIR}/math/mathmore/inc)
```

Comment:
MathCore must not depend on MathMore. Is this really a bug in our code?!

Reply:
Seems to be, it was missing lib dependency. Now it is not required and I removed it. Let see if it still compiled on all platforms

Reply:
The dependency should not be at library level. It is true for convenience the files Math/PdfFunc.h and similar include, if R__HAS_MATHMORE is defines, the math more headers, but this file is only for users. It should not be used inside Mathcore, also for the dictionary, If it is used it is a mistake.
So there is no need to also add Mathmore in the list of headers for building MathCore

Reply:
To compile ROOT, I have to add `MathMore` includes to `MathCore`. 
Just checking usage of `Math/DistFunc.h` include file.
It is used in `mathcore/test/testRootFinder.cxx` and in `mathcore/src/RandomFunctions.cxx`.
`Math/QuantFunc.h` used in dictionary generation
`Math/SpecFunc.h` used for many tests is mathcore
To resolve this - one need extra PR.

Reply:
@linev https://github.com/root-project/root/pull/5418 Lorenzo opened PR, so you can drop this change...

Reply:
Once Lorenzo PR merged - I will drop this place. 
I think, there are several tests which should be fixed as well

Reply:
Also in several sources MathMore is used - PR will not help at all